tests/framework/population/test_manager.py

```diff
@@ -154,6 +155,73 @@ def test_get_population(
 
 
+def test_get_population_different_attribute_types() -> None:
+    """Test that get_population works with simple attributes, non-simple attributes,
+    and attribute pipelines that return dataframes instead of series'."""
+    component = AttributePipelineCreator()
+    sim = InteractiveContext(components=[component], setup=True)
+    pop = sim._population.get_population("all", untracked=True)
+    # We have columnar multi-index due to AttributePipelines that return dataframes
+    assert isinstance(pop.columns, pd.MultiIndex)
+    assert set(pop.columns) == {
+        ("tracked", ""),
+        ("simulant_step_size", ""),
+        ("test_column_1", ""),
+        ("test_column_2", ""),
+        ("test_column_3", ""),
+        ("attribute_generating_columns_4_5", "test_column_4"),
+        ("attribute_generating_columns_4_5", "test_column_5"),
+        ("test_attribute", ""),
+        ("attribute_generating_columns_6_7", "test_column_6"),
+        ("attribute_generating_columns_6_7", "test_column_7"),
+    }
+    value_cols = [
+        col for col in pop.columns if not col in [("tracked", ""), ("simulant_step_size", "")]
+    ]
+    expected = pd.Series([idx % 3 for idx in pop.index])
+    for col in value_cols:
+        pd.testing.assert_series_equal(pop[col], expected, check_names=False)
+
+
+def test_get_population_column_ordering() -> None:
+    def _extract_ordered_list(cols: list[str]) -> list[tuple[str, str]]:
+        col_mapping = {
+            "tracked": ("tracked", ""),
+            "test_column_1": ("test_column_1", ""),
+            "attribute_generating_columns_4_5": [
+                ("attribute_generating_columns_4_5", "test_column_4"),
+                ("attribute_generating_columns_4_5", "test_column_5"),
+            ],
+            "test_attribute": ("test_attribute", ""),
+        }
+        expected_cols = []
+        for col in cols:
+            mapping = col_mapping[col]
+            if isinstance(mapping, list):
+                expected_cols.extend(mapping)
+            else:
+                expected_cols.append(mapping)
+        return expected_cols
+
+    def _check_col_ordering(sim: InteractiveContext, cols: list[str]) -> None:
+        pop = sim._population.get_population(cols, untracked=True)
+        expected_cols = _extract_ordered_list(cols)
+        assert isinstance(pop.columns, pd.MultiIndex)
+        returned_cols = pop.columns.tolist()
+        assert returned_cols == expected_cols
+
+    component = AttributePipelineCreator()
+    sim = InteractiveContext(components=[component], setup=True)
+
+    cols = ["tracked", "test_column_1", "attribute_generating_columns_4_5", "test_attribute"]
+    _check_col_ordering(sim, cols)
+    # Now try reversing the order
+    # NOTE: we specifically do not parameterize this test to ensure that the two
+    # 'get_population' calls are happening on exaclty the same population manager
+    cols.reverse()
+    _check_col_ordering(sim, cols)
```

Comment:
I'm surprised this passes. I wonder if we got lucky with our small number of columns?

Reply:
Was the include duplicates parameterization failing prior to your fix?